components/esp_lvgl_port/CMakeLists.txt

```diff
@@ -119,5 +123,15 @@ target_link_libraries(lvgl_port_lib PRIVATE
     ${ADD_LIBS}
     )
 
+if (NOT button_ver)
+    message("IOT button version is not set or empty!")
+    set(button_ver "4.0.0")
+endif()
+
+#Get component button major version
+string(REPLACE "." ";" BUTTON_VERSION_LIST ${button_ver})
+list(GET BUTTON_VERSION_LIST 0 button_ver_major)
+target_compile_definitions(lvgl_port_lib PUBLIC "-DCOMPONENT_BUTTON_VERSION_MAJOR=${button_ver_major}")
```

Comment:
The version define should come from the button component 

https://github.com/espressif/esp-iot-solution/blob/4730d91db70df7e6e0a3191d725ab1c5f98ff9ce/tools/cmake_utilities/package_manager.cmake#L44

So it should be `BUTTON_VER_MAJOR`. Could you please try it? It would help us avoid any convoluted CMake logic

Reply:
@tore-espressif What do you think about backward compatibility of BSP? Now, there is no breaking change of API, but it is not compatible with old button component. Should it be compatible with both? 
Other BSPs will be updated in next PR.